Framework/DataHandling/src/SaveDiffFittingAscii.cpp

```diff
@@ -0,0 +1,246 @@
+#include <fstream>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace Kernel;
+using namespace Mantid::API;
+
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(SaveDiffFittingAscii)
+
+using namespace Kernel;
+using namespace API;
+
+/// Empty constructor
+SaveDiffFittingAscii::SaveDiffFittingAscii()
+    : Mantid::API::Algorithm(), m_sep(','), m_endl('\n'), m_counter(0) {}
+
+/// Initialisation method.
+void SaveDiffFittingAscii::init() {
+
+  declareProperty(make_unique<WorkspaceProperty<ITableWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "The name of the workspace containing the data you want to "
+                  "save to a TBL file");
+
+  // Declare required parameters, filename with ext {.his} and input
+  // workspace
+  const std::vector<std::string> exts{".txt", ".csv", ""};
+  declareProperty(Kernel::make_unique<API::FileProperty>(
+                      "Filename", "", API::FileProperty::Save, exts),
+                  "The filename to use for the saved data");
+
+  declareProperty(
+      "RunNumber", "", boost::make_shared<MandatoryValidator<std::string>>(),
+      "Run number list of the focused files, which is used to generate the "
+      "parameters table workspace");
+
+  declareProperty(
+      "Bank", "", boost::make_shared<MandatoryValidator<std::string>>(),
+      "Bank number list of the focused files, which is used to generate "
+      "the parameters table workspace");
+
+  std::vector<std::string> formats;
+
+  formats.push_back("AppendToExistingFile");
+  formats.push_back("WriteGroupWorkspace");
+  formats.push_back("OverwriteFile");
+  declareProperty(
+      "OutFormat", "AppendToExistingFile",
+      boost::make_shared<Kernel::StringListValidator>(formats),
+      "Append data to existing file or save multiple table workspaces"
+      "in a group workspace");
+}
+
+/**
+*   Executes the algorithm.
+*/
+void SaveDiffFittingAscii::exec() {
+
+  // Retrieve the input workspace
+  /// Workspace
+  const ITableWorkspace_sptr tbl_ws = getProperty("InputWorkspace");
+  if (!tbl_ws)
+    throw std::runtime_error("Please provide an input workspace to be saved.");
+
+  m_workspaces.push_back(
+      boost::dynamic_pointer_cast<DataObjects::TableWorkspace>(tbl_ws));
+
+  processAll();
+}
+
+bool SaveDiffFittingAscii::processGroups() {
+  try {
+
+    std::string name = getPropertyValue("InputWorkspace");
+
+    WorkspaceGroup_sptr inputGroup =
+        AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(name);
+
+    for (int i = 0; i < inputGroup->getNumberOfEntries(); ++i) {
+      m_workspaces.push_back(
+          boost::dynamic_pointer_cast<ITableWorkspace>(inputGroup->getItem(i)));
+    }
+
+    // Store output workspace in AnalysisDataService
+    if (!isChild())
+      this->store();
+
+    setExecuted(true);
+    notificationCenter().postNotification(
+        new FinishedNotification(this, this->isExecuted()));
```

Comment:
This code (L99-L105) should be removed in the future (hopefully soon) when the issue of the finalization of algorithms in `processGroups()` is fixed (#16838).